internal/nginx/maxmind.go

```diff
@@ -186,12 +185,12 @@ func downloadDatabase(dbName string) error {
 				return err
 			}
 
-			defer outFile.Close()
-
 			if _, err := io.CopyN(outFile, tarReader, header.Size); err != nil {
 				return err
 			}
 
+			outFile.Close()
```

Comment:
Was this move some linter complain? I'm afraid we never reach this close if some error occurs before

Reply:
The error is: `deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop (gocritic)`

Reply:
I think we can revert the change here, not sure if there is a better solution.

Reply:
So, the bug deal here is that in a case of copy failing and returning error, we will never close the file.

Suggestion is: create a function just for this case, with the first statement on os.Create, and if no err after that doing a defer outFile.Close()

But for the sake of this PR, do it on another one :)

Reply:
What I mean in this case is, revert it for now, and let's fix it on a followup PR :) You can already open an issue for it if you want and work on it, let's just not mix things right now

Reply:
Revert it, and open a followup issue: https://github.com/kubernetes/ingress-nginx/issues/10299